src/Compilers/CSharp/Portable/Compilation/CSharpDiagnosticFilter.cs

```diff
@@ -146,15 +147,27 @@ public static Diagnostic Filter(Diagnostic d, int warningLevelOption, NullableCo
                 report = isEnabledByDefault ? ReportDiagnostic.Default : ReportDiagnostic.Suppress;
             }
 
+            var position = location.SourceSpan.Start;
+
             bool isNullableFlowAnalysisWarning = ErrorFacts.NullableFlowAnalysisWarnings.Contains(id);
+            if (isNullableFlowAnalysisWarning)
```

Comment:
>if (isNullableFlowAnalysisWarning) [](start = 12, length = 34)

I suspect this fixes [https://github.com/dotnet/roslyn/issues/32742](https://github.com/dotnet/roslyn/issues/32742)

Reply:
What do you think of moving the isNullableFlowAnalysisWarning handling at the start of this method?
As it stands, we're computing `report` above this block, but we're checking it below this block, which is weird.

---
In reply to: [294555110](https://github.com/dotnet/roslyn/pull/36464#discussion_r294555110) [](ancestors = 294555110)

Reply:
I think you're right, it might as well occur before step 1.

---
In reply to: [295034304](https://github.com/dotnet/roslyn/pull/36464#discussion_r295034304) [](ancestors = 295034304,294555110)

Reply:
Could you add some tests to confirm whether you've fixed [https://github.com/dotnet/roslyn/issues/32742](https://github.com/dotnet/roslyn/issues/32742) with this PR?

Scenario:
```C#
#nullable enable annotations
class Base { virtual void M(List<string> s) { } }
class C : Base { override void M(List<string?> s) { } } // expecting no nullability warning since nullable warnings context is false, even though the relevant OHI warning (CS8610) isn't produced by NullableWalker
```

---
In reply to: [295034304](https://github.com/dotnet/roslyn/pull/36464#discussion_r295034304) [](ancestors = 295034304,294555110)

Reply:
I'm adding a few tests however I believe this isn't resolved by this PR because it's simply implemented by checking the `ImmutableHashSet<ErrorCode> NullableFlowAnalysisWarnings`.

---
In reply to: [295043412](https://github.com/dotnet/roslyn/pull/36464#discussion_r295043412) [](ancestors = 295043412,295034304,294555110)